File: Web/src/api/groupUser.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetGroupUserList(data) {
  return apiClient.post("/groupUser/GetGroupUserList", data).then(handleResponse).catch(handleError)
}

function AddGroupUser(data) {
    return apiClient.post("/groupUser/AddGroupUser", data).then(handleResponse).catch(handleError)
}

function RemoveGroupUser(data) {
    return apiClient.post("/groupUser/RemoveGroupUser", data).then(handleResponse).catch(handleError)
}

const groupUserApi = {
  GetGroupUserList,
  AddGroupUser,
  RemoveGroupUser
}

export default groupUserApi

File: Server/route/groupUser.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetGroupUserRoute(r *gin.Engine) {
	adminGroupUser := r.Group("/groupUser")
	adminGroupUser.Use(middleware.AdminAuthMiddleware())
	{
		adminGroupUser.POST("/GetGroupUserList", api.GetGroupUserList)
		adminGroupUser.POST("/AddGroupUser", api.AddGroupUser)
		adminGroupUser.POST("/RemoveGroupUser", api.RemoveGroupUser)
	}
}
